test/test_optim.py

```diff
@@ -750,6 +818,16 @@ def test_multi_tensor_optimizers(self):
         ]
         self._test_derived_optimizers(optimizer_pairs_with_flags, "foreach")
 
+    @unittest.skipIf(not TEST_MULTIGPU, "only one GPU detected")
+    def test_multi_tensor_optimizers_with_varying_tensors(self):
+        optimizer_pairs_with_flags = [
+            (optim.Adadelta, dict(weight_decay=0)),
+            (optim.Adadelta, dict(weight_decay=1)),
+            (optim.Adagrad, dict(weight_decay=0)),
+            (optim.Adagrad, dict(weight_decay=1)),
```

Comment:
Why are we not testing all the optimizers here?

Reply:
Cuz the majority of them just fall back to the slow implementation anyway and it's more interesting to add these tests when I change their implementations to use the group by util.

Reply:
I added the majority anyway. Will add NAdam when the time comes.